subsys/bluetooth/controller/ll_sw/ull_llcp_phy.c

```diff
@@ -110,6 +186,88 @@ static uint16_t lp_event_counter(struct ll_conn *conn)
 	return event_counter;
 }
 
+static inline uint8_t util_instant_passed(struct ll_conn *conn, uint16_t instant)
+{
+	// if instant already passed, quit procedure with error */
+	uint16_t event_counter = util_event_counter(conn);
+	if (((event_counter - instant) & 0xFFFF) <= 0x7FFF) {
+		return 1;
+	}
+	return 0;
+}
+
+static uint8_t util_check_update_ind(struct ll_conn *conn, struct proc_ctx *ctx)
+{
+	/* Both tx and rx PHY unchanged */
+	if (!((ctx->data.pu.m_to_s_phy | ctx->data.pu.s_to_m_phy) & 0x07)) {
+		// if no phy changes, quit procedure, and possibly signal host
+		ctx->data.pu.error = BT_HCI_ERR_SUCCESS;
+		return 1;
+	} else {
+		// if instant already passed, quit procedure with error */
+		if (util_instant_passed(conn, ctx->data.pu.instant)) {
+			ctx->data.pu.error = BT_HCI_ERR_INSTANT_PASSED;
+			return 1;
+		}
+	}
+	return 0;
+}
+
+static uint8_t util_apply_phy_update(struct ll_conn *conn, struct proc_ctx *ctx)
+{
+	if (ctx->data.pu.s_to_m_phy) {
+		if (conn->lll.role == BT_HCI_ROLE_SLAVE) {
+			conn->lll.phy_tx = ctx->data.pu.s_to_m_phy;
+		} else {
+			conn->lll.phy_rx = ctx->data.pu.s_to_m_phy;
+		}
+	}
+
+	if (ctx->data.pu.m_to_s_phy) {
+		if (conn->lll.role == BT_HCI_ROLE_SLAVE) {
+			conn->lll.phy_rx = ctx->data.pu.m_to_s_phy;
+		} else {
+			conn->lll.phy_tx = ctx->data.pu.m_to_s_phy;
+		}
+	}
+
+	return (ctx->data.pu.m_to_s_phy || ctx->data.pu.s_to_m_phy);
+}
+
+static inline void util_set_preferred_phys(struct ll_conn *conn, uint8_t tx, uint8_t flags, uint8_t rx)
+{
+	conn->phy_pref_rx = rx;
+	conn->phy_pref_tx = tx;
+	conn->lll.phy_flags = flags;
```

Comment:
I inherited this too, and I had the same thought but decided after looking into it a while back, that it is indeed correct. However when I look at it now, I can't seem to remember why I thought it was ok! phy_pref_flags was introduced by me when I thought I needed it, so either I do need it or it should be removed again. 
Perhaps we can discuss tomorrow?

Reply:
Ahh, I remembered again - sleep is good:
Since 'flags' indicate local coded phy preference (S2 or S8) and this is not negotiated with the peer, it is simply re-configured when the update is initiated, and takes effect whenever the coded phy is in use. So all is good and I remove the 'void' phy_pref_flags entry

Reply:
Add a comment in the code about this

Reply:
added comment